### PR TITLE
Fix path mappings with remote debugging

### LINKED ISSUE #610: Source references not working
## Environment data

- PTVSD version: master
- OS and version: windows to windows
- Python version (& distribution if applicable, e.g. Anaconda): 3.7
- Using VS Code or Visual Studio: VSC

## Actual behavior

Pausing while using remote only file (no path mapping) will bring up:
image

## Expected  behavior

Should use source reference to pull the remote file

### On remote machine:
1. Create this file `remote_test.py`:
```python
import time
import ptvsd

ptvsd.enable_attach()

i = 0
while True:
    print(i)
    time.sleep(1)
    i += 1
```
2. Ensure port `5678` has access for inbound connections.
3. run `py -3.7 remote_test.py`

### On dev machine:
1.  Use following launch config:
```json
        {
            "name": "PyExp: Attach (Remote)",
            "type": "pythonExperimental",
            "request": "attach",
            "port": 5678,
            "host": "<remote-pc-name>",
            "logToFile": true
        },
```
2. Select `PyExp: Attach (Remote)` to attach.
3. Hit pause after attaching.

### PR DESCRIPTION
**Pull request by DonJayamanne**
Fixes #610

@karthiknadig Please get this PR locally and test at your end.
I have tested from 
* Mac -> Win
* Win -> Mac
* Mac -> Mac
* Win -> Win

Please test this specific scenario https://github.com/Microsoft/ptvsd/issues/610#issuecomment-403618960

Note: I'm not sure `~/GIT` will work, you might want to use absolute paths.

I'll be working on system tests.

### CODE REVIEW

**Comment by karthiknadig** [comment]
Tested this and it fixes the mapping issue.

**Comment by karthiknadig** [comment]
The wrapper.py change looks good. I see that you are still working on tests. Let me know when they are done.

File: ptvsd/wrapper.py [hunk 1]
```diff
@@ -1419,34 +1421,34 @@ def _should_debug(self, filepath):
...
+        if len(self._path_mappings) > 0:
+            pydevd_file_utils.setup_client_server_paths(self._path_mappings)
 
     def _send_cmd_version_command(self):
         cmd = pydevd_comm.CMD_VERSION
-        windows_client = self.debug_options.get(
-            'WINDOWS_CLIENT',
-            platform.system() == 'Windows')
-        os_id = 'WINDOWS' if windows_client else 'UNIX'
+        default_os_type = 'WINDOWS' if sys.platform == 'win32' else 'UNIX'
```
Comment (karthiknadig) [inline ptvsd/wrapper.py]: >sys.platform == 'win32' [](start = 39, length = 23)

platform.system()

File: README.md [hunk 2]
```diff
@@ -45,7 +45,8 @@ contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additio
             "Jinja",                // Enables Jinja (Flask) Template debugging
             "FixFilePathCase",      // See FIX_FILE_PATH_CASE in wrapper.py
             "DebugStdLib"           // Whether to enable debugging of standard library functions
-            "WindowsClient"         // Whether client OS is Windows or not
+            "WindowsClient"         // Whether client OS is Windows
+            "UnixClient"            // Whether client OS is Unix
```
Comment (karthiknadig) [inline README.md]: >UnixClient [](start = 13, length = 10)

This should be updated.

Comment (DonJayamanne) [reply README.md]: It is, this is what VSC sends,

File: ptvsd/wrapper.py (hunk 1 above)
Comment (DonJayamanne) [reply ptvsd/wrapper.py]: oops, thanks

**Pull request merged**